WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsActivity.java

```diff
@@ -80,366 +26,45 @@ public void onCreate(Bundle savedInstanceState) {
-                                                mNotesListFragment.setNoteIsHidden(note.getId(), false);
-                                                ToastUtils.showToast(NotificationsActivity.this,
-                                                        R.string.error_moderate_comment,
-                                                        ToastUtils.Duration.LONG
-                                                );
-                                            }
-                                        }
-                                    });
-                        }
-
-                        @Override
-                        public void onClear(@Nonnull Parcelable[] token) {
-                            //noop
-                        }
-
-                        @Override
-                        public void onUndo(Parcelable parcelable) {
-                            mNotesListFragment.setNoteIsHidden(note.getId(), false);
-                        }
-                    }).show();
-        }
-    }
-
-    @Override
-    public void onCommentChanged(CommentActions.ChangedFrom changedFrom, CommentActions.ChangeType changeType) {
-        // pop back stack if we edited a comment notification, so simperium will show the change
-        if (changedFrom == CommentActions.ChangedFrom.COMMENT_DETAIL
-                && changeType == CommentActions.ChangeType.EDITED) {
-            FragmentManager fm = getFragmentManager();
-            if (fm.getBackStackEntryCount() > 0) {
-                fm.popBackStack();
-            }
-        }
-    }
-
-    private class NoteClickListener implements NotificationsListFragment.OnNoteClickListener {
-        @Override
-        public void onClickNote(String noteId) {
-            if (TextUtils.isEmpty(noteId)) return;
-
-            // open the latest version of this note just in case it has changed - this can
-            // happen if the note was tapped from the list fragment after it was updated
-            // by another fragment (such as NotificationCommentLikeFragment)
-            if (SimperiumUtils.getNotesBucket() != null) {
-                try {
-                    Note note = SimperiumUtils.getNotesBucket().get(noteId);
-                    openNote(note);
-                } catch (BucketObjectMissingException e) {
-                    AppLog.e(T.NOTIFS, "Note could not be found.");
-                }
-            }
-        }
-    }
-
-    // Retrieves comment reply text so we can restore it upon returning to CommentDetailFragment
-    private String getCommentReplyText() {
-        if (mDetailFragment != null && mDetailFragment instanceof CommentDetailFragment) {
-            CommentDetailFragment commentDetailFragment = (CommentDetailFragment)mDetailFragment;
-            if (!TextUtils.isEmpty(commentDetailFragment.getReplyText())) {
-                return commentDetailFragment.getReplyText();
-            }
+        Intent intent = getIntent();
```

Comment:
Hmm, after working on this I've remembered that we currently use `onNewIntent` to open the notifications activity with the new note if a user opens a push while notifications are already visible. So we kinda need to process the arguments in the activity still and tell the list fragment to open the note.

Open to any ideas :)